### PR TITLE
Evolve before fight

### PR DESCRIPTION
**Pull request by bmorel**
evolve before fight, only if enough hp

### CODE REVIEW

File: bots/default_aliens.bt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 selector
 {
-	condition ( aliveTime > 1500 )
+	condition ( aliveTime > 1500 && healScore < 0.5 )
```
Comment (slipher) [inline bots/default_aliens.bt:3]: healScore involves distance to friendly structures as well as HP. Is that intentional?

Comment (bmorel) [reply bots/default_aliens.bt:3]: It is, but yeah, it's arguable, same for the score I used.

Comment (bmorel) [reply bots/default_aliens.bt:3]: @Slipher: I could change it so that it depends on health, instead.

Comment (slipher) [reply bots/default_aliens.bt:3]: Just curious, fine with me either way

**Comment by necessarily-equal** [comment]
This should help base defense as this lets bots evolve when close to enemies, but not in line of sight. Thanks!

**Comment by bmorel** [comment]
Yep, the BT changes I propose are all based on ideas I got while observing bots endlessly :)

**Pull request merged**